### DISCUSSION TITLE
How can I use directives or `$wire` inside a named slot?

### DISCUSSION THREAD

**Discussion by johnRivs** [livewire/livewire#8524] [1. Help] [answered]
Is there a way to make directives or `$wire` work inside a named slot?

This is my layout file:
```blade
<body>
    {{ $header }}

    {{ $slot }}
</body>
```

And this is the Volt component:
```blade
<?php

use function Livewire\Volt\state;

state(['email' => 'user@example.com']);

$update = function() {
    // ...
};

?>

<x-slot:header>
    <!-- This won't work. -->
    <button wire:click="update">Update {{ $email }}</button>
    <button x-data @click="$wire.update()">Update {{ $email }}</button>
</x-slot>

<div>
    <!-- Livewire stuff will work here. -->
</div>
```

**Comment by joshhanley** [comment, marked as answer]
@johnRivs you can't Livewire directives need to be contained within a Livewire component. By added them to the slot, Blade moves them out of the component, hence they won't work.

**Reply by johnRivs** [reply]
Is there a clever workaround? So far, I've been doing this:
```blade
<button onclick="Livewire.getByName('component.name')[0].update()"></button>
```

**Reply by Dimimo** [reply]
A bit late to react but it caught my attention. It reminds me to a similar problem I encountered recently. 

Does your remark include blade templates? as in 
`<x-page :data="$data"/>`? 
It seems to me it works, as long it's not more than one blade template level deep. If this is the case, I goes haywire after a few updates (even with wire:key) resulting in page not found. A hack is to create random keys and force the livewire component to re-render. But... yes, I know, that's why it's called a hack. 

My suspicion goes to blade templates. A blade template inside of a blade template seems to lose its ability to communicate up and down. Even with emitting events from blade to livewire and visa versa. 

Or I do it wrong and should rely on livewire templates instead of blade templates.

**Comment by jethromayuk** [comment]
While its probably not ideal, I simply created a Header component and added that to the slot, moving events there. It works for my particular need.

**Reply by johnRivs** [reply]
Could you show a brief example?
